File: src/test/java/com/github/qky666/selenidepom/test/java/mtp/pom/plainjavapom/ServicesPage.java
package com.github.qky666.selenidepom.test.java.mtp.pom.plainjavapom;

import com.codeborne.selenide.SelenideElement;
import com.github.qky666.selenidepom.pom.Required;

import java.time.Duration;

import static com.codeborne.selenide.Condition.text;
import static com.codeborne.selenide.Selenide.$;

public class ServicesPage extends MainFramePage {
    @Required public final SelenideElement principal = $("div.servicios-principal");
    @Required public final SelenideElement title = $("h1.h2");
    public final SelenideElement badSelector = $("bad-selector");

    @Override
    public void customShouldLoadRequired(Duration timeout, String model, String lang) throws Throwable {
        super.customShouldLoadRequired(timeout, model, lang);
        title.shouldHave(text("Aseguramiento de la calidad"), timeout);
    }
}

File: cookiecutter/{{cookiecutter.repo_name}}/modules/mtptestng/src/test/kotlin/{{ cookiecutter.__group_folder }}/mtptestng/pom/pages/services/ServicesPage.kt
package {{ cookiecutter.group }}.mtptestng.pom.pages.services

import com.github.qky666.selenidepom.pom.Required
import {{ cookiecutter.group }}.mtptestng.pom.common.MainFramePage

open class ServicesPage : MainFramePage() {
    @Required open val title = find("div.servicios-principal h1.h2")
}

val servicesPage = ServicesPage()
